app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -20,51 +20,106 @@ import android.annotation.SuppressLint
 import android.arch.lifecycle.Observer
 import android.arch.lifecycle.ViewModelProviders
 import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
 import android.view.inputmethod.EditorInfo
 import android.widget.TextView
 import com.duckduckgo.app.global.DuckDuckGoActivity
-import kotlinx.android.synthetic.main.activity_main.*
-import timber.log.Timber
+import com.duckduckgo.app.global.ViewModelFactory
+import com.duckduckgo.app.global.view.hide
+import com.duckduckgo.app.global.view.hideKeyboard
+import com.duckduckgo.app.global.view.show
+import kotlinx.android.synthetic.main.activity_browser.*
+import kotlinx.android.synthetic.main.content_browser.*
 import javax.inject.Inject
 
 class BrowserActivity : DuckDuckGoActivity() {
 
     @Inject lateinit var webViewClient: BrowserWebViewClient
-    @Inject lateinit var viewModelFactory: BrowserViewModel.BrowserViewModelFactory
+    @Inject lateinit var webChromeClient: BrowserChromeClient
+    @Inject lateinit var viewModelFactory: ViewModelFactory
 
     private val viewModel: BrowserViewModel by lazy {
         ViewModelProviders.of(this, viewModelFactory).get(BrowserViewModel::class.java)
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContentView(R.layout.activity_browser)
 
-        viewModel.query.observe(this, Observer {
-            if (savedInstanceState == null) {
-                Timber.v("Webview loading url $it")
-                webView.loadUrl(it)
-            }
+        viewModel.viewState.observe(this, Observer<BrowserViewModel.ViewState> {
+            it?.let { render(it) }
         })
 
-        loadUrlButton.setOnClickListener({
-            userEnteredQuery()
+        viewModel.query.observe(this, Observer {
+            it?.let { webView.loadUrl(it) }
         })
 
+        configureSwipeToRefresh()
+        configureToolbar()
         configureWebView()
+        configureUrlInput()
+    }
+
+    private fun render(it: BrowserViewModel.ViewState) {
```

Comment:
I see `it` as  param shortening for simple lambdas, similar to using `i` in a for loop. Do we want it here too? I feel a more expressive name would work better.

Reply:
👍 

only reason it is called `it` is because of IDE refactoring that into its own method. definitely good call for better name here